conda-store-server/conda_store_server/plugins/lock/conda_lock.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) conda-store development team. All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+import json
+import os
+import pathlib
+import typing
+
+import yaml
+from conda_lock.conda_lock import run_lock
+
+from conda_store_server._internal import conda_utils, schema, utils
+from conda_store_server.plugins import hookspec
+from conda_store_server.plugins.plugin_context import PluginContext
+
+
+class CondaLock:
+    @classmethod
+    def name(cls):
+        return "lock-conda_lock"
+
+    def __init__(
+        self, conda_flags="--strict-channel-priority", conda_command="mamba", *kwargs
+    ):
+        self.conda_command = conda_command
+        self.conda_flags = conda_flags
+
+    @utils.run_in_tempdir
+    @hookspec.hookimpl
```

Comment:
Hm, I was expecting the implementations to live in a different abstraction level, not so close to the actual code. For reference, this is [how conda does it](https://github.com/conda/conda/blob/cf3a0fa9ce01ada7a4a0c934e17be44b94d4eb91/conda/plugins/solvers.py#L9-L17).